src/mock-doc/test/selector.spec.ts

```diff
@@ -225,4 +226,50 @@ describe('selector', () => {
     const q2 = div.querySelector('span.c a');
     expect(q2.tagName).toBe('A');
   });
+
+  it.each(PROBLEMATIC_SELECTORS)("should error for '%p' selector", (selector) => {
+    const doc = new MockDocument();
+
+    const expectedMessage = [
+      `At present jQuery does not support the ${selector} selector.`,
+      'If you need this in your test, consider writing an end-to-end test instead.',
+      `Syntax error, unrecognized expression: unsupported pseudo: ${selector.replace(':', '')}`,
+    ].join('\n');
+
```

Comment:
Can we add an assertion for `matches` as well?
```suggestion

    expect(() => doc.matches(selector)).toThrow(expectedMessage);
```

Reply:
oh good idea 👍